app/views/two_factor_authentication/webauthn_verification/show.html.erb

```diff
@@ -8,7 +8,15 @@
   <%= @presenter.webauthn_help %>
 </div>
 
-<%= form_tag(login_two_factor_webauthn_path, method: :patch, class: 'margin-bottom-1 read-after-submit', id: 'webauthn_form') do %>
+<%= simple_form_for(
+      '',
+      url: login_two_factor_webauthn_path,
+      method: :patch,
+      html: {
+        class: 'margin-bottom-1 read-after-submit',
+        id: 'webauthn_form',
+      },
+    ) do |f| %>
   <%= hidden_field_tag :user_challenge,
                        '[' + user_session[:webauthn_challenge].split.join(',') + ']', id: 'user_challenge' %>
```

Comment:
Wonder if the `split` is even necessary? From what I can tell, the value should already be assigned as an array.

Reply:
It must be converted to a string... [because arrays don't have `#split` ... ](https://ruby-doc.org/core-3.0.0/Array.html)

Reply:
> It must be converted to a string... [because arrays don't have `#split` ... ](https://ruby-doc.org/core-3.0.0/Array.html)

It was an array when I threw a `pry` in there. The `split` comes from Rails:

https://guides.rubyonrails.org/active_support_core_extensions.html#split-value-nil

Reply:
😱 yikes, that was not the TIL I was looking for, thanks

Reply:
Updated in cfa7aff.